parsons/catalist/catalist.py

```diff
@@ -0,0 +1,388 @@
+            static_values,
+        )
+        result = self.await_completion(response["id"])
+        return result
+
+    def upload(
+        self,
+        table: Table,
+        export: bool = False,
+        description: Optional[str] = None,
+        export_filename_suffix: Optional[str] = None,
+        input_subfolder: Optional[str] = None,
+        copy_to_sandbox: bool = False,
+        static_values: Optional[Dict[str, Union[str, int]]] = None,
+    ) -> dict:
+        """Load table to the Catalist Match API, returns response with job metadata.
+
+        `Args:`
+             export: bool
+                 Defaults to False
+             description: str
+                 Optional. Description for the match job.
+             export_filename_suffix: str
+                 Optional. Adds a suffix to the result filename in the SFTP server.
+             input_subfolder: str
+                 Optional. Adds a prefix to the filepath of the uploaded input file in
+                 the SFTP server.
+             copy_to_sandbox: bool
+                  Defaults to False.
+             static_values: dict
+                  Optional. Any included values are mapped to every row of the input table.
+        """
+
+        self.validate_table(table)
+
+        # upload table to s3 temp location
+        sftp_file_path = self.load_table_to_sftp(table, input_subfolder)
+        sftp_file_path_encoded = base64.b64encode(
+            sftp_file_path.encode("ascii")
+        ).decode("ascii")
+
+        if export:
+            action = "export%2Cpublish"
+        else:
+            action = "publish"
+
+        # Create endpoint using options
+        endpoint_params = [
+            "upload",
+            "template",
+            str(template_id),
+            "action",
+            action,
+            "url",
+            sftp_file_path_encoded,
+        ]
+
+        if description:
+            endpoint_params.extend(["description", description])
+
+        endpoint = "/".join(endpoint_params)
+
+        # Assemble query parameters
+        query_params: Dict[str, Union[str, int]] = {"token": self.token}
+        if copy_to_sandbox:
+            query_params["copyToSandbox"] = "true"
+        if static_values:
+            query_params.update(static_values)
+        if export_filename_suffix:
+            query_params["subClientName"] = export_filename_suffix
+
+        logger.debug(f"Executing request to endpoint {self.connection.uri + endpoint}")
+
+        endpoint = endpoint + "?" + urllib.parse.urlencode(query_params)
+
+        response = self.connection.get_request(endpoint)
+
+        result = response[0]
+
+        return result
+
+    def action(
+        self,
+        file_ids: Union[str, List[str]],
+        match: bool = False,
+        export: bool = False,
+        export_filename_suffix: str = "",
+        copy_to_sandbox: bool = False,
+    ) -> List[dict]:
+        """Perform actions on existing files.
```

Comment:
Missing args in docstring

Reply:
Done with 561deccce2f6a219c0270c382c2906805ecdbd75